compiler/rustc_middle/src/ty/print/pretty.rs

```diff
@@ -904,14 +906,21 @@ pub trait PrettyPrinter<'tcx>:
                     first = false;
                 }
 
-                for (assoc_item_def_id, ty) in assoc_items {
+                for (assoc_item_def_id, term) in assoc_items {
+                    let ty = match term.skip_binder() {
+                        Term::Ty(ty) => ty,
+                        Term::Const(c) => {
+                            p!(print(c));
+                            continue;
+                        }
+                    };
```

Comment:
you need to do this near the `match ty.kind()` below, and you can't use `continue` as that will skip the `first = false`